src/parsers/intersection.ts

```diff
@@ -19,7 +20,16 @@ export function parseIntersectionDef(
       ...refs,
       currentPath: [...refs.currentPath, "allOf", "1"],
     }),
-  ].filter((x): x is JsonSchema7Type => !!x);
+  ].filter((x): x is JsonSchema7Type => !!x)
+    .map((schema) => {
+      if ('additionalProperties' in schema && schema.additionalProperties === false) {
+        const {additionalProperties, ...rest} = schema;
+        return {
+          ...rest
+        };
+      }
+      return schema
+    });
 
-  return allOf.length ? { allOf } : undefined;
+  return allOf.length ? {allOf, unevaluatedProperties: false} : undefined;
```

Comment:
Only set this to false if the conditional on line 25 kicks in. You could instantiate it as undefined beforehand and set it to false in the if-statement, for instance

Reply:
Actually I guess all of the object schemas should have additionalProperties set to false for this to apply